Sources/CoreLocationMiddleware/CoreLocationMiddleware.swift

```diff
@@ -1,50 +1,62 @@
 import CoreLocation
 import SwiftRex
 
-public enum LocationState {
-    case unknown
-    case notAuthorized
-    case authorized(lastPosition: CLLocation?)
+public struct LocationState: Equatable {
+    var authzType: AuthzType
```

Comment:
Thanks !

Reply:
It won't be very frequent, but it can if the application chooses "when-in-use" first and then for some reason wants to upgrade to "always". In that case the user will be asked one more time (with a different message) to confirm the authorization type (see screenshot below). Does that make sense ? 

![Simulator Screen Shot - iPhone 11 - 2020-09-23 at 19 00 08](https://user-images.githubusercontent.com/3268395/94085907-04466000-fdcf-11ea-9a50-1afbf7baa018.png)